File: app/src/main/java/co/tpcreative/supersafe/common/controller/GalleryCameraMediaManager.java
package co.tpcreative.supersafe.common.controller;
public class GalleryCameraMediaManager {
    private static GalleryCameraMediaManager instance;
    public boolean isProgressing() {
        return isProgressing;
    }

    public void setProgressing(boolean progressing) {
        isProgressing = progressing;
    }

    private boolean isProgressing;

    public static GalleryCameraMediaManager getInstance(){
        if (instance==null){
            instance = new GalleryCameraMediaManager();
        }
        return instance;
    }
}

File: app/src/main/java/co/tpcreative/supersafe/common/controller/GalleryCameraMediaManager.kt
package co.tpcreative.supersafe.common.controller

class GalleryCameraMediaManager {
    fun isProgressing(): Boolean {
        return isProgressing
    }

    fun setProgressing(progressing: Boolean) {
        isProgressing = progressing
    }

    private var isProgressing = false

    companion object {
        private var instance: GalleryCameraMediaManager? = null
        fun getInstance(): GalleryCameraMediaManager? {
            if (instance == null) {
                instance = GalleryCameraMediaManager()
            }
            return instance
        }
    }
}